File: app/schemas/conversation.py
from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field


class ConversationCreateRequest(BaseModel):
    project_name: str = Field(..., min_length=1)


class ConversationCreateResponse(BaseModel):
    conversation_id: uuid.UUID


class ConversationMessageRead(BaseModel):
    id: uuid.UUID
    conversation_id: uuid.UUID
    role: Literal["USER", "ASSISTANT"]
    content: str
    trace_json: dict[str, Any] | None
    created_at: datetime


class ConversationRead(BaseModel):
    id: uuid.UUID
    project_name: str
    created_at: datetime
    updated_at: datetime
    messages: list[ConversationMessageRead] = Field(default_factory=list)

File: ui/src/types/conversation.ts
import type { IncidentSearchResult } from './answer'
import type { AgentTrace } from './trace'

export type ConversationRole = 'USER' | 'ASSISTANT'

export type ConversationMessageRead = {
  id: string
  conversation_id: string
  role: ConversationRole
  content: string
  trace_json?: AgentTrace | null
  created_at: string
}

export type ConversationRead = {
  id: string
  project_name: string
  created_at: string
  updated_at: string
  messages: ConversationMessageRead[]
}

export type ChatMessageModel = {
  id: string
  role: ConversationRole
  content: string
  createdAt: string
  trace?: AgentTrace | null
  incidents?: IncidentSearchResult[]
}
